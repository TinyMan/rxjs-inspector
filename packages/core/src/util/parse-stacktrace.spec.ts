import { parseStackTrace } from '.';

describe('parseStacktrace', () => {
  it('should correlty parse a stack trace', () => {
    const fixture =
      'Error: 0\n    at MapSubscriber.project (http://localhost:1337/main.js:641:19)\n    at MapSubscriber.push.../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (http://localhost:1337/vendor.js:61785:35)\n    at MapSubscriber.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)\n    at Wrapper.push.../packages/core/src/patch/subscribe.ts.Wrapper._next (http://localhost:1337/main.js:230:51)\n    at Wrapper.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)\n    at TakeSubscriber.push.../node_modules/rxjs/_esm5/internal/operators/take.js.TakeSubscriber._next (http://localhost:1337/vendor.js:65380:30)\n    at TakeSubscriber.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)\n    at Wrapper.push.../packages/core/src/patch/subscribe.ts.Wrapper._next (http://localhost:1337/main.js:230:51)\n    at Wrapper.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)\n    at AsyncAction.dispatch [as work] (http://localhost:1337/vendor.js:56400:14)';
    expect(parseStackTrace(fixture)).toMatchSnapshot();
  });
});
