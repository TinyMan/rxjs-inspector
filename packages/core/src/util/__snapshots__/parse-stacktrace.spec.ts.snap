// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseStacktrace should correlty parse a stack trace 1`] = `
Object {
  "col": "19",
  "file": "http://localhost:1337/main.js",
  "line": "641",
  "stack": Array [
    "Error: 0",
    "    at MapSubscriber.project (http://localhost:1337/main.js:641:19)",
    "    at MapSubscriber.push.../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (http://localhost:1337/vendor.js:61785:35)",
    "    at MapSubscriber.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)",
    "    at Wrapper.push.../packages/core/src/patch/subscribe.ts.Wrapper._next (http://localhost:1337/main.js:230:51)",
    "    at Wrapper.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)",
    "    at TakeSubscriber.push.../node_modules/rxjs/_esm5/internal/operators/take.js.TakeSubscriber._next (http://localhost:1337/vendor.js:65380:30)",
    "    at TakeSubscriber.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)",
    "    at Wrapper.push.../packages/core/src/patch/subscribe.ts.Wrapper._next (http://localhost:1337/main.js:230:51)",
    "    at Wrapper.push.../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (http://localhost:1337/vendor.js:53584:12)",
    "    at AsyncAction.dispatch [as work] (http://localhost:1337/vendor.js:56400:14)",
  ],
}
`;
